File: app/src/main/java/net/bloople/manga/PageFragment.java
package net.bloople.manga;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.fragment.app.Fragment;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;

public class PageFragment extends Fragment {
    private Context context;
    private MangosUrl url;

    static PageFragment newInstance(MangosUrl url) {
        PageFragment fragment = new PageFragment();
        Bundle args = new Bundle();
        args.putParcelable("url", url);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        this.context = context;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        url = getArguments().getParcelable("url");
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.page_fragment, container, false);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        ImageView imageView = view.findViewById(R.id.image);

        Glide
                .with(this)
                .load(url.toGlideUrl())
                .transform(new MatchWidthTransformation())
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .into(imageView);
    }

    @Override
    public void onDetach() {
        super.onDetach();
        this.context = null;
    }
}

File: app/src/main/java/net/bloople/manga/PageFragment.kt
package net.bloople.manga

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.fragment.app.Fragment
import coil3.load

class PageFragment : Fragment() {
    private lateinit var url: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        url = requireArguments().getString("url")!!
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.page_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val imageView: ImageView = view.findViewById(R.id.image)

        imageView.load(url)
    }

    companion object {
        fun newInstance(url: MangosUrl): PageFragment {
            return PageFragment().apply {
                arguments = Bundle().apply {
                    putString("url", url.toString())
                }
            }
        }
    }
}